src/application/page1/ReservationUnitModal.tsx

```diff
@@ -125,63 +132,156 @@ const Text = styled.span`
   font-size: var(--fontsize-heading-s);
 `;
 
-const StyledSearchInput = styled(SearchInput)`
-  margin-top: 2em;
+const Filters = styled.div`
+  @media (max-width: var(--breakpoint-s)) {
+    grid-template-columns: 1fr;
+  }
+
+  margin-top: var(--spacing-m);
+  display: grid;
+  grid-template-columns: 1fr 1fr;
+  gap: var(--spacing-m);
+`;
+
+const ButtonContainer = styled.div`
+  margin-top: var(--spacing-m);
+  display: flex;
+  align-items: center;
+`;
+
+const SearchButton = styled(Button).attrs({
+  type: 'submit',
+})`
+  margin-right: var(--spacing-m);
+`;
+
+const Ruler = styled.hr`
+  margin-top: var(--spacing-layout-m);
 `;
 
 const Results = styled.div`
-  height: 60vh;
-  overflow-y: auto;
+  margin-bottom: 112px;
 `;
 
+const StyledLoadingSpinner = styled(LoadingSpinner).attrs({ small: true })``;
+
+type OptionsType = {
+  purposeOptions: OptionType[];
+  reservationUnitTypeOptions: OptionType[];
+};
+
+const emptyOption = {
+  label: '',
+};
+
 const ReservationUnitModal = ({
   applicationPeriod,
   handleAdd,
   currentReservationUnits,
+  options,
 }: {
   applicationPeriod: ApplicationPeriod;
   handleAdd: (ru: ReservationUnit) => void;
   currentReservationUnits: ReservationUnit[];
+  options: OptionsType;
 }): JSX.Element => {
-  const [q, setQ] = useState<string | null>(null);
+  const [searchTerm, setSearchTerm] = useState<string | undefined>(undefined);
   const [selected, setSelected] = useState<ReservationUnit[]>([]);
+  const [purpose, setPurpose] = useState<OptionType | undefined>(undefined);
+  const [reservationUnitType, setReservationUnitType] = useState<
+    OptionType | undefined
+  >(undefined);
+  const [results, setRes] = useState<ReservationUnit[] | [] | undefined>(
+    undefined
+  );
+  const [searching, setSearching] = useState<boolean>(false);
+
+  const purposeOptions = [emptyOption].concat(options.purposeOptions);
+  const reservationUnitTypeOptions = [emptyOption].concat(
+    options.reservationUnitTypeOptions
+  );
 
   const { t } = useTranslation();
 
-  const results = useAsync(async () => {
-    if (q === null) {
-      return [];
-    }
-    return getReservationUnits({ search: q });
-  }, [q]);
+  const searchResults = async () => {
+    setSearching(true);
+    const searchCriteria = {
```

Comment:
What about the ApplicationPeriod filtering?

Reply:
Forgot about that - added.

Reply:
Not super important right now, but the api also supports filtering with parameter application_period , for example application_period=1